<launch>
  <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="launch_prefix" default="" />

  <!-- configure camera input -->
  <arg name="camera_name" default="/usb_cam" />
  <arg name="image_topic" default="image_raw" />
  <arg name="queue_size" default="1" />
  <include file="$(find usb_cam)/launch/usb_cam-test.launch">
  </include>
  <!-- apriltag_ros continuous detection node -->
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros_continuous_node" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)">
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />

    <param name="publish_tag_detections_image" type="bool" value="true" /><!-- default: false -->
    <param name="queue_size" type="int" value="$(arg queue_size)" />

    <!-- load parameters (incl. tag family, tags, etc.) -->
    <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml"/>
    <rosparam command="load" file="$(find apriltag_ros)/config/tags.yaml"/>
  </node>
  <!--将转换后的坐标发给飞控 -->
  <node pkg="apriltag_ros" type="tftomavros" name="tftomavros" output="screen" >
  </node>
  <!--将识别的结果图像发送至qgc-->
  <node pkg="apriltag_ros" type="imgtoqgc" name="imgtoqgc" output="screen" >
    <param name="udp_host" value="127.0.0.1" />
    <param name="img_width" type="int" value="640"/>
    <param name="img_height" type="int" value="480"/>
  </node>
</launch>
